langchain4j-workers-ai/src/main/java/dev/langchain4j/model/workersai/WorkersAiEmbeddingModel.java

```diff
@@ -0,0 +1,248 @@
+         */
+        public String apiToken;
+        /**
+         * ModelName, preferred as enum for extensibility.
+         */
+        public String modelName;
+
+        /**
+         * Simple constructor.
+         */
+        public Builder() {
+        }
+
+        /**
+         * Simple constructor.
+         *
+         * @param accountId
+         *      account identifier.
+         * @return
+         *      self reference
+         */
+        public Builder accountId(String accountId) {
+            this.accountId = accountId;
+            return this;
+        }
+
+        /**
+         * Sets the apiToken for the Worker AI model builder.
+         *
+         * @param apiToken The apiToken to set.
+         * @return The current instance of {@link WorkersAiChatModel.Builder}.
+         */
+        public Builder apiToken(String apiToken) {
+            this.apiToken = apiToken;
+            return this;
+        }
+
+        /**
+         * Sets the model name for the Worker AI model builder.
+         *
+         * @param modelName The name of the model to set.
+         * @return The current instance of {@link WorkersAiChatModel.Builder}.
+         */
+        public Builder modelName(String modelName) {
+            this.modelName = modelName;
+            return this;
+        }
+
+        /**
+         * Builds a new instance of Worker AI Chat Model.
+         *
+         * @return A new instance of {@link WorkersAiChatModel}.
+         */
+        public WorkersAiEmbeddingModel build() {
+            return new WorkersAiEmbeddingModel(this);
+        }
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public Response<Embedding> embed(String text) {
+        try {
+            dev.langchain4j.model.workersai.client.WorkersAiEmbeddingRequest req = new dev.langchain4j.model.workersai.client.WorkersAiEmbeddingRequest();
+            req.getText().add(text);
+
+            retrofit2.Response<dev.langchain4j.model.workersai.client.WorkersAiEmbeddingResponse> retrofitResponse = workerAiClient
+                    .embed(req, accountId, modelName)
+                    .execute();
+
+            processErrors(retrofitResponse.body(), retrofitResponse.errorBody());
+            if (retrofitResponse.body() == null) {
+                throw new RuntimeException("Unexpected response: " + retrofitResponse);
+            }
+            dev.langchain4j.model.workersai.client.WorkersAiEmbeddingResponse.EmbeddingResult res = retrofitResponse.body().getResult();
+            // Single Vector expected
+            if (res.getShape().get(0) != 1) {
+                throw new RuntimeException("Unexpected shape: " + res.getShape());
+            }
+            List<Float> embeddings = res.getData().get(0);
+            float[] floatArray = new float[embeddings.size()];
+            for (int i = 0; i < embeddings.size(); i++) {
+                floatArray[i] = embeddings.get(i); // Unboxing Float to float
```

Comment:
minor: you could change the type from List<Float> to float[] in `res.getData()` class and avoid doing conversion manually